src/assets/TokenListController.ts

```diff
@@ -279,6 +279,13 @@ export class TokenListController extends BaseController<
           fetchTokenList(this.chainId, this.abortController.signal),
         );
         if (!tokensFromAPI) {
```

Comment:
Is it possible that a failure in fetching the token list could cause `!tokensFromAPI` to be true when on mainnet or another supported network, and then we could be clearing the token list when instead we should be relying on the tokenList already in state?